seshdash/views.py

```diff
@@ -898,33 +902,123 @@ def edit_site(request,site_Id=1):
     context_dict['sites_stats'] = sites_stats
     return render(request,'seshdash/settings.html', context_dict)
 
+
+
+@login_required
+@permission_required_or_403('auth.view_Sesh_Site')
+def site_add_edit(request):
+    """
+    This views renders the page for editing and
+    adding new rmc sites
+    """
+    context_dict = {}
+    sites = _get_user_sites(request)
+    context_dict['sites_stats'] = get_quick_status(sites)
+    context_dict['VRM_form'] = _create_vrm_login_form()
+    return render(request, 'seshdash/settings.html', context_dict)
+    
+
+
 # function of adding new site
 @login_required
 @permission_required_or_403('auth.view_Sesh_Site')
-def add_site(request):
+def add_rmc_site(request):
+    """
+    This adds an rmc site to 
+    the database
+    """
     context_dict = {}
-    #checking permissions
-    user = request.user
-    permission = get_permissions(user)
 
-    #fetching list of sites for the user
-    user_sites =  _get_user_sites(request)
-    sites_stats = get_quick_status(user_sites)
-    # on ajax
     if request.method == 'POST':
-
         form = SiteForm(request.POST)
 
         if form.is_valid():
             form = form.save()
-            form = SiteForm()
+            return HttpResponseRedirect(reverse('add_rmc_account', args=[form.id]))
 
-    #on page load
     else:
         form = SiteForm()
 
-    context_dict['permitted'] = permission
-    context_dict['sites_stats'] = sites_stats
-    context_dict['form_add'] = form
+    context_dict['form'] = form
+    return render(request, 'seshdash/add_rmc_site.html', context_dict)
+
+
+
+@login_required
+@permission_required_or_403('auth.view_Sesh_Site')
+def add_rmc_account(request, site_id):
+    """
+    This view is for adding a new rmc account
+    to the database and associate it with a site
+    """
+    site = Sesh_Site.objects.filter(id=site_id).first()
+
```

Comment:
`sensor_arr = ['EmonTh', 'EmonTx', 'BMV']
fomset_dict = {}
for sensor in sensor_arr:
     formset_dict[sensor] = formset_factory('Sensor'+sensor+'Form')

     
`